### PR TITLE
DOCSP-33631: multi-field join tutorial

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-33631>
Staging - https://preview-mongodbrustagir.gatsbyjs.io/node/DOCSP-33631-multi-field-join/aggregation-tutorials/multi-field-join/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/aggregation-tutorials/multi-field-join.txt [hunk 1]
```diff
@@ -4,4 +4,240 @@
...
+
+This example uses two collections:
+
+- ``products``, which contains documents describing the products that
+  a shop sells
+- ``orders``, which contains documents describing individual orders
+  for products in a shop
+
+An order can only contain one product, so the aggregation uses a
+multi-field join to match a product document to documents for orders of
```
Comment (norareidy) [inline source/aggregation-tutorials/multi-field-join.txt:56]: S: I think "for" is a little unclear
```suggestion
multi-field join to match a product document to documents representing orders of
```

File: source/includes/aggregation/multi-field-join.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        category: "KITCHENWARE",
+        description: "Luxury mixer turning good cakes into great",
+      },
+      // end-products
+    ];
+
+    const sampleData2 = [
+      // start-orders
+      {
+        customer_id: "elise_smith@myemail.com",
```
Comment (norareidy) [inline source/includes/aggregation/multi-field-join.js:53]: S: The style guide recommends using "example.com" for emails (applies to the other emails too)

```suggestion
        customer_id: "elise_smith@example.com",
```

File: source/aggregation-tutorials/multi-field-join.txt [hunk 3]
```diff
@@ -4,4 +4,240 @@
...
+               :copyable: true
+               :start-after: start-unset
+               :end-before: end-unset
+               :dedent:
+
+   .. step:: Run the Aggregation Pipeline
+
+      Perform the aggregation on the ``products`` collection by running the
+      following command in your shell to run the ``agg_tutorial.js``
+      file:
```
Comment (norareidy) [inline source/aggregation-tutorials/multi-field-join.txt:194]: S: Not necessary, but I found the current wording slightly confusing
```suggestion
      Run the `agg_tutorial.js`` file to perform the aggregation on the ``products`` collection by running the
      following command in your shell:
```

File: source/aggregation-tutorials/multi-field-join.txt [hunk 4]
```diff
@@ -4,4 +4,240 @@
...
+           orders: [
+             {
+               customer_id: 'oranieri@warmmail.com',
+               orderdate: 2020-01-01T08:25:37.000Z,
+               value: 63.13
+             }
+           ]
+         }
+
+      The result consists of documents that contain details from
```
Comment (norareidy) [inline source/aggregation-tutorials/multi-field-join.txt:238]: S: just to shorten
```suggestion
      The result documents contain details from
```

File: source/aggregation-tutorials/multi-field-join.txt [hunk 5]
```diff
@@ -4,4 +4,240 @@
...
+
+.. meta::
+   :keywords: code example, node.js, lookup, aggregation
+
+Introduction
+------------
+
+In this tutorial, you can learn how to use the {+driver-short+} to
+perform a multi-field join. A multi-field join occurs when there are
+multiple fields in the documents of two collections that you use to
```
Comment (norareidy) [inline source/aggregation-tutorials/multi-field-join.txt:25]: S: I think adding "corresponding" clarifies why/how you can use these multiple fields to match documents together

```suggestion
multiple corresponding fields in the documents of two collections that you use to
```

File: source/aggregation-tutorials/multi-field-join.txt [hunk 6]
```diff
@@ -4,4 +4,240 @@
...
+            .. literalinclude:: /includes/aggregation/multi-field-join.js
+               :language: javascript
+               :copyable: true
+               :start-after: start-embedded-pl-unset
+               :end-before: end-embedded-pl-unset
+               :dedent:
+
+            After the embedded pipeline is completed, add the
+            ``$lookup`` stage to the main aggregation pipeline that also stores
+            the processed lookup fields in the ``orders`` array field:
```
Comment (norareidy) [inline source/aggregation-tutorials/multi-field-join.txt:154]: S: breaking up the sentence might help make it more action-oriented

```suggestion
            ``$lookup`` stage to the main aggregation pipeline. Configure this stage to store
            the processed lookup fields in the ``orders`` array field:
```

**Review by norareidy — CHANGES_REQUESTED** [review]
Good work, I added a few suggestions!

File: source/aggregation-tutorials/multi-field-join.txt (hunk 3 above)
Comment (rustagir) [reply source/aggregation-tutorials/multi-field-join.txt:194]: Simplified and changed in both existing tutorials

File: source/includes/aggregation/multi-field-join.js (hunk 2 above)
Comment (norareidy) [reply source/includes/aggregation/multi-field-join.js:53]: Ah okay you can ignore this suggestion then!

**Pull request merged**